src/main/kotlin/com/openlattice/mechanic/upgrades/CleanOutOrgMembersAndRoles.kt

```diff
@@ -0,0 +1,55 @@
+package com.openlattice.mechanic.upgrades
+
+import com.openlattice.authorization.Ace
+import com.openlattice.authorization.Acl
+import com.openlattice.authorization.AuthorizationManager
+import com.openlattice.authorization.Permission
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.organizations.roles.SecurePrincipalsManager
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class CleanOutOrgMembersAndRoles(
+        private val toolbox: Toolbox,
+        private val principalsManager: SecurePrincipalsManager,
+        private val authorizationManager: AuthorizationManager
+) : Upgrade {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(CleanOutOrgMembersAndRoles::class.java)
+        private val MEMBER_PERMISSIONS = EnumSet.of(Permission.READ)
+    }
+
+    override fun upgrade(): Boolean {
+        val organizationsMapstore = HazelcastMap.ORGANIZATIONS.getMap(toolbox.hazelcast)
+
+        val organizations = organizationsMapstore.toMap()
+        val principals = HazelcastMap.PRINCIPALS.getMap(toolbox.hazelcast).values.associate { it.principal to it.aclKey }
+
+        organizations.values.forEach {
+            logger.info("About to clean out members for org {} [{}]", it.title, it.id)
+
+            val memberAclKeys = it.members.mapNotNull { p -> principals[p] }.toSet()
+            principalsManager.addPrincipalToPrincipals(it.getAclKey(), memberAclKeys)
+            authorizationManager.addPermissions(listOf(
+                    Acl(
+                            it.getAclKey(),
+                            it.members.map { p -> Ace(p, MEMBER_PERMISSIONS) }
+                    )
+            ))
+
+            it.members.clear()
+            it.roles.clear()
+            organizationsMapstore[it.id] = it
+
+            logger.info("Finished cleaning out {} members for org {} [{}]", memberAclKeys.size, it.title, it.id)
+        }
+
+        return true
+    }
+
+    override fun getSupportedVersion(): Long {
+        return Version.V2020_10_14.value
+    }
```

Comment:
What's this?

Reply:
We use hardcoded versions to specify which set of migrations will run when we run mechanic. Basically it'll run whichever version you specify so you label all migrations you want to run with the same version